src/main/java/com/unstoppabledomains/resolution/contracts/ens/Registry.java

```diff
@@ -0,0 +1,35 @@
+package com.unstoppabledomains.resolution.contracts.ens;
+
+import java.io.IOException;
+
+import com.unstoppabledomains.resolution.contracts.Contract;
+
+public class Registry extends Contract {
+
+  private static final String ABI_FILE = "src/main/resources/abi/ens_registry_abi.json";  
+
+  public Registry(String url, String address) {
+    super(url, address, ABI_FILE);
+  }
+
+  public String getResolverAddress(byte[] tokenId) {
+    Object[] args = new Object[1];
+    args[0] = tokenId;
+    try {
+      return this.fetchAddress("resolver", args);
+    } catch(IOException exception) {
```

Comment:
@JohnnyJumper Should throw an error in case of IOException instead of silently return null?

Reply:
This was done on purpose to throw an error on level above. I can probably make it throw NS exception with something like blockchain is down, but I am not sure if that is a good idea over here